doc/RELEASE.adoc

```diff
@@ -0,0 +1,120 @@
+
+We do not tag or publish alpha or beta verions for libraries. Libraries are published on Hackage, and Hackage does not support version qualifiers. It is also uncommon for a library to publish pre-release versions (unless it is new or a complete rewrite).
+However, we may annotate individual functions in a library as alpha or beta if we deem it appropriate.
+
+We do tag release candidates, e.g,. 3.5.2-rc1, because additional non-CI tests and checks may need to be carried out.
+These tests and checks either are too expensive to run in CI, or require manual steps (e.g., security audit), but are prerequisites of making a release.
+Tagging release candidates makes it easy to identify the correct commit to run these tests and checks against.
+
+For executables, we may publish alpha, beta or release candidates at our discretion.
+
+== Release Frequency
+
+We will begin by making a new major or minor release every 4 weeks, and evaluate and adjust the frequency in the future.
+Ad-hoc releases can be made upon request.
+
+We only make patch releases for version `x.y`, if a cardano-node release candidate depends on `x.y.z`, and an issue is found during the node release process that requires patching plutus.
+In all other cases, we always start a new major or minor release from master.
+
+== Release Process
+
+=== Major and Minor Releases
+
+Suppose we are releasing version `x.y.0`.
+
+1. Update version numbers in cabal files to `x.y.0` (if it is not already the case).
+- Version bounds for packages in the same repo should be set to `==x.y.*`.
+2. Make sure the change log is up to date.
+3. Tag `x.y.0-rc1` on master.
+4. Run all tests and checks.
+- This includes all tests that can be done with only the Plutus repo, such as unit tests, property-based tests, conformance tests and nightly tests.
+- This does not include tests that require building a new node, or integrating with other downstream projects.
+- In some cases we need input from domain experts and/or security audit (e.g., when adding a cryptography builtin). The right group of people should be involved and sign off on the tests and checks.
+5. If no release blocking issue is found, do a formal release and upload the packages to Hackage.
+- If issues are found, create a release branch `release/x.y`, fix the issues on master, backport the fixes to `release/x.y`, tag `x.y.0-rc2`, and go to step 4.
+- Why not just fix the issues on master and tag `x.y.0-rc2` from master?
+It is desirable to minimize the amount of change between `rc1` and `rc2`, because it may reduce the tests and checks that need to be performed against `rc2`.
+For instance, if `plutus-ledger-api` is the only package changed, there is no need to re-run tests on `plutus-core` or `plutus-tx`.
+Another example is if a security audit is done on `rc1`, and the changes in `rc2` does not modify the audited code, then the audit does not need to be re-done.
+
+6. After the release, update version numbers in cabal files to `x.y+1.0`.
```

Comment:
This is a tricky corner. AIUI, there are two main strategies for bumping version numbers:
1. Bump them when you make a change that requires a bump. So if you do something that requires a new major version, you bump to the next major version. But this requires you to remember whether you've done a release since the last time you bumped - if you have you don't bump.
2. Bump them when you want to do a release. This requires you to go back through your changes and decide what kind of bump is merited.

It seems like you're suggesting a kind of hybrid of 1 and 2: the first point of this process made me think we're doing 2, but this makes me think that we're doing 1. Maybe worth discussing on its own! Personally I think 1 is probably fine, it's not that hard to remember what the last released version was so long as you only have one!

Reply:
I think 1 can be cumbersome. If we haven't bumped the major version, then every PR needs to be evaluated to see whether it should cause the major version to be bumped. I think we can go with 2, but instead of going back through the changes, just bump the major version (second PVP component, unless the change is big enough to justify bumping the first), unless we are certain that we only need to bump the minor version. What do you think?

Reply:
I'm happy with either tbh! I think it would be nice to call this out as it's own thing, since I predict "when do I bump the versions?" to be a regular question!